src/Cms/ModelWithContent.php

```diff
@@ -260,11 +260,15 @@ public function isValid(): bool
     /**
      * Returns the lock object for this model
      *
-     * @return Kirby\Cms\ContentLock
+     * @return Kirby\Cms\ContentLock|null
      */
     public function lock()
     {
-        return new ContentLock($this);
+        // only return object if directory exist,
+        // virtual pages will need to overwrite this method
+        if (file_exists($this->contentFileDirectory()) === true) {
```

Comment:
This should first check if the return value is a string. Calling `file_exists()` with a non-string param is undefined behavior.